### PR TITLE
Extension note should be amended, to reflect this is also used for shortening a booking

### PR DESCRIPTION
**Pull request by bryangaledxw**
# Changes in this PR

When displaying booking info, the text "Extension notes", will now either read "Extension notes" or "Shortening notes", depending on whether the booking has been extended or shortened

### Before
localhost_3000_properties_b6a87a7e-1b6a-4c96-b7ff-8c0dc414796d_bedspaces_467deeb6-33ca-4d34-8713-60faaf230318_bookings_90100146-1a0a-4500-aef5-947c90472f5f (2)
localhost_3000_properties_b6a87a7e-1b6a-4c96-b7ff-8c0dc414796d_bedspaces_467deeb6-33ca-4d34-8713-60faaf230318_bookings_90100146-1a0a-4500-aef5-947c90472f5f_history (2)

### After
localhost_3000_properties_b6a87a7e-1b6a-4c96-b7ff-8c0dc414796d_bedspaces_467deeb6-33ca-4d34-8713-60faaf230318_bookings_90100146-1a0a-4500-aef5-947c90472f5f (1)

localhost_3000_properties_b6a87a7e-1b6a-4c96-b7ff-8c0dc414796d_bedspaces_467deeb6-33ca-4d34-8713-60faaf230318_bookings_90100146-1a0a-4500-aef5-947c90472f5f_history (1)

# Release checklist

As part of our continuous deployment strategy we must ensure that this work is
ready to be released at any point. Before merging to `main` we must first
confirm:

## Pre merge checklist

- [ ] Have all changes to any dependencies been deployed to both `preprod` and
    `production` in a backwards compatible way? (This includes our API,
    infrastructure, third-party integrations and any secrets or environment variables)
- [ ] Has a required data migration been included in this change or been done in
    advance?

## Post merge checklist

Once we've merged we now need to release this through to production before
considering it done.

[Find the build-and-deploy job in CircleCI](https://app.circleci.com/pipelines/github/ministryofjustice/hmpps-temporary-accommodation-ui)
and work through the following steps:

- [ ] Has the product manager asked to provide approval for this feature?
  - [ ] Has the product manager approved?
- [ ] Manually approve release to preprod
- [ ] Verify change released to preprod
- [ ] Manually approve release to prod
- [ ] Verify change released to production

Should a release fail at any step, you as the author should now lead the work to
fix it as soon as possible.

### CODE REVIEW

**Comment by bryangaledxw** [comment]
[Extension note should be amended, to reflect this is also used for shortening a booking](https://trello.com/c/xU0vaDTC/794-extension-note-should-be-amended-to-reflect-this-is-also-used-for-shortening-a-booking)

File: server/utils/bookingUtils.ts [hunk 1]
```diff
@@ -117,6 +117,13 @@ export const deriveBookingHistory = (booking: Booking) => {
   return history
 }
 
+export const shortenedOrExtended = (extension: Extension): 'shortened' | 'extended' => {
+  const previousDepartureDate = DateFormats.convertIsoToDateObj(extension.previousDepartureDate)
+  const newDepartureDate = DateFormats.convertIsoToDateObj(extension.newDepartureDate)
+
+  return previousDepartureDate.getTime() > newDepartureDate.getTime() ? 'shortened' : 'extended'
```
Comment (tahb) [inline server/utils/bookingUtils.ts:124]: Shame we have to convert to date and then into time for the comparison. I assume we can't go straight into a time object.

**Review by tahb — APPROVED** [review]
LGTM!

File: server/utils/bookingUtils.ts (hunk 1 above)
Comment (bryangaledxw) [reply server/utils/bookingUtils.ts:124]: I had a quick look into this - _in theory_, if we can guarantee there will never be a timezone marker, we can sort the date strings lexically. But that feels unsafe and I'd rather err on the side of caution here

**Pull request merged**